### PR TITLE
Fixing the func(*args) issues with sympy.vector

### PR DESCRIPTION
**Pull request by srjoglekar246**
An attempt to fix the compatibility issues of sympy.vector with the core.

### CODE REVIEW

**Comment by asmeurer** [comment]
It looks like the test_args tests are failing.

**Comment by asmeurer** [comment]
Note that I intend to merge this with my fixes at https://github.com/sympy/sympy/pull/10084 (which make BaseScalar subclass Expr instead of Symbol).

**Comment by asmeurer** [comment]
You can't use None in the .args. 

An idea I had was to make points without parents only have two args. The code would then distinguish them by the number of args. This would already work with the existing constructor for `point.func(*point.args)` because of the `parent_point=None`. You would need to make sure other code handles it correctly, though.

**Comment by srjoglekar246** [comment]
Fixed the issue. Tests seem to pass on my machine now. We don't need to pass parent as an arg if its not None anyways, so I just put an 'if' statement over it.

**Comment by asmeurer** [comment]
It looks good to me. As long as you are confident that changing the .args like this won't break any other code (and obviously as long as the test pass) I am +1 to merge this.

File: sympy/vector/scalar.py [hunk 1]
```diff
@@ -12,6 +13,9 @@ class BaseScalar(Symbol):
     """
 
     def __new__(cls, name, index, system, pretty_str, latex_str):
+        name = str(name)
+        pretty_str = str(pretty_str)
+        latex_str = str(latex_str)
```
Comment (asmeurer) [inline sympy/vector/scalar.py]: I'm going to undo this change in my merge because I think it's the wrong thing in Python 2 (since pretty_str could be unicode).
